test/unit-test/ota_cbor_utest.c

```diff
@@ -132,3 +134,224 @@ void test_OTA_CborDecodeStreamResponse()
+        1,                                 /* file id. */
+        OTA_FILE_BLOCK_SIZE,               /* block size. */
+        0,                                 /* block offset. */
+        ( uint8_t * ) &bitmap,             /* block bitmap. */
+        sizeof( bitmap ),                  /* size of bitmap. */
+        otaconfigMAX_NUM_BLOCKS_REQUEST ); /* number of block requested. */
+
+    TEST_ASSERT_FALSE( result );
+
+    result = OTA_CBOR_Encode_GetStreamRequestMessage(
+        cborWork,                          /* output message buffer. */
+        sizeof( cborWork ),                /* size of output message buffer. */
+        NULL,                              /* size of encoded message. */
+        CBOR_TEST_CLIENTTOKEN_VALUE,       /* client token. */
+        1,                                 /* file id. */
+        OTA_FILE_BLOCK_SIZE,               /* block size. */
+        0,                                 /* block offset. */
+        ( uint8_t * ) &bitmap,             /* block bitmap. */
+        sizeof( bitmap ),                  /* size of bitmap. */
+        otaconfigMAX_NUM_BLOCKS_REQUEST ); /* number of block requested. */
+
+    TEST_ASSERT_FALSE( result );
+
+    result = OTA_CBOR_Encode_GetStreamRequestMessage(
+        cborWork,                          /* output message buffer. */
+        sizeof( cborWork ),                /* size of output message buffer. */
+        &encodedSize,                      /* size of encoded message. */
+        NULL,                              /* client token. */
+        1,                                 /* file id. */
+        OTA_FILE_BLOCK_SIZE,               /* block size. */
+        0,                                 /* block offset. */
+        ( uint8_t * ) &bitmap,             /* block bitmap. */
+        sizeof( bitmap ),                  /* size of bitmap. */
+        otaconfigMAX_NUM_BLOCKS_REQUEST ); /* number of block requested. */
+
+    TEST_ASSERT_FALSE( result );
+}
+
+void test_OTA_CborDecodeStreamResponse_Invalid()
+{
+    uint8_t blockPayload[ OTA_FILE_BLOCK_SIZE ] = { 0 };
+    uint8_t cborWork[ CBOR_TEST_MESSAGE_BUFFER_SIZE ] = { 0 };
+    size_t encodedSize = 0;
+    int fileId = -1;
+    int blockIndex = -1;
+    int blockSize = -1;
+    uint8_t decodedPayload[ OTA_FILE_BLOCK_SIZE ] = { 0 };
+    uint8_t * pDecodedPayload = decodedPayload;
+    size_t payloadSize = -1;
+    bool result = false;
+    bool msgValidity = false;
+    int i = 0;
+
+    /* Test OTA_CBOR_Decode_GetStreamResponseMessage( ). */
+    for( i = 0; i < ( int ) sizeof( blockPayload ); i++ )
+    {
+        blockPayload[ i ] = i % UINT8_MAX;
+    }
+
+    result = createOtaStreammingMessage(
+        cborWork,
+        sizeof( cborWork ),
+        CBOR_TEST_BLOCKIDENTITY_VALUE,
+        blockPayload,
+        sizeof( blockPayload ),
+        &encodedSize,
+        msgValidity );
+    TEST_ASSERT_EQUAL( CborNoError, result );
+
+    if( TEST_PROTECT() )
```

Comment:
Why are these TEST_PROTECT statements being used? I don't see any calls to TEST_ABORT.

Reply:
These were for debugging the failing tests, removed them in https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/150/commits/f6b16a4e84acc2710323a793bb8513a2811d7632